Scripts/Pointer/BasePointer.cs

```diff
@@ -0,0 +1,298 @@
+﻿namespace VRTK.Core.Pointer
+{
+    using UnityEngine;
+    using UnityEngine.Events;
+    using System;
+    using VRTK.Core.Utility;
+    using VRTK.Core.Data.Type;
+
+    /// <summary>
+    /// The PointerData contains information about the current pointer state.
+    /// </summary>
+    public class PointerData : SurfaceData
+    {
+        /// <summary>
+        /// Determines if the pointer is currently in it's activated state.
+        /// </summary>
+        public bool isActive;
+        /// <summary>
+        /// Determines if the pointer is in it's valid state.
+        /// </summary>
+        public bool isValid;
+        /// <summary>
+        /// The duration that the pointer has been hovering over it's current target.
+        /// </summary>
+        public float hoverDuration;
+    }
+
+    /// <summary>
+    /// The BasePointer forms the basis for all pointer types.
+    /// </summary>
+    public abstract class BasePointer : MonoBehaviour
+    {
+        /// <summary>
+        /// The visibility of the pointer element.
+        /// </summary>
+        public enum VisibilityType

```

Comment:
This feels usable in other scenarios, too. Should we move it in the `Type` namespace and adjust the documentation on it to be generic?

Reply:
Agreed, but maybe we should move it when there is a need for it to be in multiple places?

Reply:
OK 👍